### PR TITLE
Added 'Rocket League' command

### PR DESCRIPTION
**Pull request by YoussefExplicit**
Get's the stats of players, works on all platforms

### CODE REVIEW

**Comment by AndrewJWin** [comment]
Proof?

**Comment by MrJacz** [comment]
I don't recommend the use of Rocket league stats, their API is heavily rate limited

File: commands/games/rocket.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            description: "Get's a user's rocket league profile.",
+            category: "Games",
+            usage: "rocket [pc/xbl/psn] [username]",
+            extended: "None.",
+            cooldown: 10
+        });
+    };
+
+    async run(message, args, level) {
+        if (!args[0] || !args[1]) return message.response("❗", `Invalid Usage, please do:\`${this.help.usage}\``);
```
Comment (MrJacz) [inline commands/games/rocket.js]: use `!args.length`

File: commands/games/rocket.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if (args[0] === "pc" || args[0] === "steam") platform = "steam";
+        if (args[0] === "ps4" || args[0] === "psn") platform = "psn";
+        if (args[0] === "xbl" || args[0] === "xbox") platform = "xbl";
+
+        try {
+            rocket.getPlayer(args.splice(1).join(" "), platform).then(player => {
```
Comment (MrJacz) [inline commands/games/rocket.js]: Use async/await

File: commands/games/rocket.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                const embed = new MessageEmbed()
+                .setImage(player.signatureUrl)
+                if(player.avatar) embed.setThumbnail(player.avatar)
```
Comment (MrJacz) [inline commands/games/rocket.js]: Add this before `message.channel.send(embed)`

File: commands/games/rocket.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                embed.setURL(player.profileUrl)
+                embed.addField("Wins", player.stats.wins, true)
+                embed.addField("Goals", player.stats.goals, true)
+                embed.addField("MVPs", player.stats.mvps, true)
+                embed.addField("Saves", player.stats.saves, true)
+                embed.addField("Shots", player.stats.shots, true)
+                embed.addField("Assists", player.stats.assists, true)
+                embed.addBlankField()
+                embed.setColor(message.guild.member(client.user.id).roles.highest.color || 0x00AE86);
+                message.channel.send(embed)
```
Comment (MrJacz) [inline commands/games/rocket.js]: Missing semi colon

File: commands/games/rocket.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                embed.setTitle(`${player.displayName}, on ${player.platform.name}`)
+                embed.setURL(player.profileUrl)
+                embed.addField("Wins", player.stats.wins, true)
+                embed.addField("Goals", player.stats.goals, true)
+                embed.addField("MVPs", player.stats.mvps, true)
+                embed.addField("Saves", player.stats.saves, true)
+                embed.addField("Shots", player.stats.shots, true)
+                embed.addField("Assists", player.stats.assists, true)
+                embed.addBlankField()
+                embed.setColor(message.guild.member(client.user.id).roles.highest.color || 0x00AE86);
```
Comment (MrJacz) [inline commands/games/rocket.js]: use `message.guild.me`

**Review by MrJacz — CHANGES_REQUESTED** [review]

File: commands/games/rocket.js (hunk 1 above)
Comment (YoussefExplicit) [reply commands/games/rocket.js]: done all 👍

**Comment by YoussefExplicit** [comment]
https://youssef.hacked-your.webcam/fxjEzRKYn.png
proof

File: commands/games/rocket.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    };
+
+    async run(message, args, level) {
+        if (!args[0].length || !args[1].length) return message.response("❗", `Invalid Usage, please do:\`${this.help.usage}\``);
+        if (args[0] === "pc" || args[0] === "steam") platform = "steam";
+        if (args[0] === "ps4" || args[0] === "psn") platform = "psn";
+        if (args[0] === "xbl" || args[0] === "xbox") platform = "xbl";
+
+        try {
+            await rocket.getPlayer(args.splice(1).join(" "), platform).then((player) => {
```
Comment (MrJacz) [inline commands/games/rocket.js]: This is not how you use async/await...

**Review by MrJacz — CHANGES_REQUESTED** [review]

File: commands/games/rocket.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+const { MessageEmbed } = require("discord.js")
+const Command = require(`${process.cwd()}/base/Command.js`);
+
+class RocketLeague extends Command {
+    constructor(client) {
+        super(client, {
+            name: "rocket",
+            description: "Get's a user's rocket league profile.",
+            category: "Games",
+            usage: "rocket [pc/xbl/psn] [username]",
```
Comment (MrJacz) [inline commands/games/rocket.js]: also use `|` not `/`

File: commands/games/rocket.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            extended: "None.",
+            cooldown: 10
+        });
+    };
+
+    async run(message, args, level) {
+        if (!args[0].length || !args[1].length) return message.response("❗", `Invalid Usage, please do:\`${this.help.usage}\``);
```
Comment (MrJacz) [inline commands/games/rocket.js]: this is not what I meant either

File: commands/games/rocket.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    };
+
+    async run(message, args, level) {
+        if (!args.length || !args[1].length) return message.response("❗", `Invalid Usage, please do:\`${this.help.usage}\``);
+        if (args[0] === "pc" || args[0] === "steam") platform = "steam";
+        if (args[0] === "ps4" || args[0] === "psn") platform = "psn";
+        if (args[0] === "xbl" || args[0] === "xbox") platform = "xbl";
+
+        try {
+            const player = await rocket.getPlayer(args.splice(1).join(" "), platform)
```
Comment (MrJacz) [inline commands/games/rocket.js]: Missing semi

File: commands/games/rocket.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+                .setImage(player.signatureUrl)
+                .setTitle(`${player.displayName}, on ${player.platform.name}`)
+                .setURL(player.profileUrl)
+                .addField("Wins", player.stats.wins, true)
+                .addField("Goals", player.stats.goals, true)
+                .addField("MVPs", player.stats.mvps, true)
+                .addField("Saves", player.stats.saves, true)
+                .addField("Shots", player.stats.shots, true)
+                .addField("Assists", player.stats.assists, true)
+                .setColor(message.guild.me.roles.highest.color || 0x00AE86)
```
Comment (MrJacz) [inline commands/games/rocket.js]: Missing semi

**Review by MrJacz — CHANGES_REQUESTED** [review]

File: commands/games/rocket.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+const { MessageEmbed } = require("discord.js")
+const Command = require(`${process.cwd()}/base/Command.js`);
+
+class RocketLeague extends Command {
+    constructor(client) {
+        super(client, {
+            name: "rocket",
+            description: "Get's a user's rocket league profile.",
+            category: "Games",
+            usage: "rocket [pc|xbl|psn] [username]",
```
Comment (YorkAARGH) [inline commands/games/rocket.js]: Can you change these to required? `<pc|xb1|psn> <username>`

File: commands/games/rocket.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+                .addField("Goals", player.stats.goals, true)
+                .addField("MVPs", player.stats.mvps, true)
+                .addField("Saves", player.stats.saves, true)
+                .addField("Shots", player.stats.shots, true)
+                .addField("Assists", player.stats.assists, true)
+                .setColor(message.guild.me.roles.highest.color || 0x00AE86);
+                if(player.avatar) embed.setThumbnail(player.avatar);
+                message.channel.send(embed);
+        } catch (error) {
+            message.response("❗", `Player not found or Invalid Usage, please do:\`${this.help.usage}\``);
```
Comment (YorkAARGH) [inline commands/games/rocket.js]: Could you also console log the error as well, using the client logger.

**Review by AndrewJWin — CHANGES_REQUESTED** [review]
```
  3:47  error  Missing semicolon                              semi
   7:1   error  Expected indentation of 2 spaces but found 4   indent
   8:1   error  Expected indentation of 4 spaces but found 8   indent
   9:1   error  Expected indentation of 6 spaces but found 12  indent
  10:1   error  Expected indentation of 6 spaces but found 12  indent
  11:1   error  Expected indentation of 6 spaces but found 12  indent
  12:1   error  Expected indentation of 6 spaces but found 12  indent
  13:1   error  Expected indentation of 6 spaces but found 12  indent
  14:1   error  Expected indentation of 6 spaces but found 12  indent
  15:1   error  Expected indentation of 4 spaces but found 8   indent
  16:1   error  Expected indentation of 2 spaces but found 4   indent
  16:6   error  Unnecessary semicolon                          no-extra-semi
  18:1   error  Expected indentation of 2 spaces but found 4   indent
  18:30  error  'level' is defined but never used              no-unused-vars
  19:1   error  Expected indentation of 4 spaces but found 8   indent
  20:1   error  Expected indentation of 4 spaces but found 8   indent
  20:54  error  'platform' is not defined                      no-undef
  21:1   error  Expected indentation of 4 spaces but found 8   indent
  21:53  error  'platform' is not defined                      no-undef
  22:1   error  Expected indentation of 4 spaces but found 8   indent
  22:54  error  'platform' is not defined                      no-undef
  24:1   error  Expected indentation of 4 spaces but found 8   indent
  25:1   error  Expected indentation of 6 spaces but found 12  indent
  25:77  error  'platform' is not defined                      no-undef
  26:1   error  Expected indentation of 6 spaces but found 16  indent
  27:1   error  Expected indentation of 8 spaces but found 16  indent
  28:1   error  Expected indentation of 8 spaces but found 16  indent
  29:1   error  Expected indentation of 8 spaces but found 16  indent
  30:1   error  Expected indentation of 8 spaces but found 16  indent
  31:1   error  Expected indentation of 8 spaces but found 16  indent
  32:1   error  Expected indentation of 8 spaces but found 16  indent
  33:1   error  Expected indentation of 8 spaces but found 16  indent
  34:1   error  Expected indentation of 8 spaces but found 16  indent
  35:1   error  Expected indentation of 8 spaces but found 16  indent
  36:1   error  Expected indentation of 8 spaces but found 16  indent
  37:1   error  Expected indentation of 6 spaces but found 16  indent
  37:17  error  Expected space(s) after "if"                   keyword-spacing
  38:1   error  Expected indentation of 6 spaces but found 16  indent
  39:1   error  Expected indentation of 4 spaces but found 8   indent
  40:1   error  Expected indentation of 6 spaces but found 12  indent
  40:44  error  Missing semicolon                              semi
  41:1   error  Expected indentation of 6 spaces but found 12  indent
  42:1   error  Expected indentation of 4 spaces but found 8   indent
  42:10  error  Unnecessary semicolon                          no-extra-semi
  43:1   error  Expected indentation of 2 spaces but found 4   indent
  43:6   error  Unnecessary semicolon                          no-extra-semi
  44:2   error  Unnecessary semicolon                          no-extra-semi
```
Travis absolutely puked over this..

**Comment by YorkAARGH** [comment]
Closing this until you fix the eslint errors.

**Pull request closed without merging**